File: CS/DXApplication5/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.XtraScheduler;
using ListViewComponent;


namespace DXApplication5
{
    public partial class Form1 : XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            schedulerControl.Start = System.DateTime.Now;
            searchControl1.Client = schedulerControl;
            schedulerControl.GroupType = SchedulerGroupType.Resource;

            InitResources();
            InitAppointments();

            ListViewHelper.AddListView(schedulerControl, dateNavigator, searchControl1);
        }
        #region data
        private void InitResources()
        {
            DataHelper.FillResources(schedulerStorage, 5);
        }
        void InitAppointments()
        {
            int count = schedulerStorage.Resources.Count;
            DataHelper.GenerateEvents(schedulerStorage, count);
        }
        #endregion
        #region events
        private void searchControl1_EditValueChanged(object sender, EventArgs e) {
            ListViewHelper.SwitchToListView();
        }
        private void searchControl1_ButtonPressed(object sender, DevExpress.XtraEditors.Controls.ButtonPressedEventArgs e) {
            if(e.Button.Kind == DevExpress.XtraEditors.Controls.ButtonPredefines.Clear) {
                ListViewHelper.SwitchToNormalView();
                schedulerControl.SetSearchControl(searchControl1);
            }
        }
        
        #endregion

        private void schedulerControl_PreviewKeyDown(object sender, System.Windows.Forms.PreviewKeyDownEventArgs e)
        {
            if (e.Control && e.KeyCode == System.Windows.Forms.Keys.E)
            {
                searchControl1.Focus();
                e.IsInputKey = false;
            }
        }
    }
   
}

File: VB/DXApplication5/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.XtraScheduler
Imports ListViewComponent


Namespace DXApplication5
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
			schedulerControl.Start = System.DateTime.Now
			searchControl1.Client = schedulerControl
			schedulerControl.GroupType = SchedulerGroupType.Resource

			InitResources()
			InitAppointments()

			ListViewHelper.AddListView(schedulerControl, dateNavigator, searchControl1)
		End Sub
		#Region "data"
		Private Sub InitResources()
			DataHelper.FillResources(schedulerStorage, 5)
		End Sub
		Private Sub InitAppointments()
			Dim count As Integer = schedulerStorage.Resources.Count
			DataHelper.GenerateEvents(schedulerStorage, count)
		End Sub
		#End Region
		#Region "events"
		Private Sub searchControl1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles searchControl1.EditValueChanged
			ListViewHelper.SwitchToListView()
		End Sub
		Private Sub searchControl1_ButtonPressed(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.ButtonPressedEventArgs) Handles searchControl1.ButtonPressed
			If e.Button.Kind = DevExpress.XtraEditors.Controls.ButtonPredefines.Clear Then
				ListViewHelper.SwitchToNormalView()
				schedulerControl.SetSearchControl(searchControl1)
			End If
		End Sub

		#End Region

		Private Sub schedulerControl_PreviewKeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.PreviewKeyDownEventArgs) Handles schedulerControl.PreviewKeyDown
			If e.Control AndAlso e.KeyCode = System.Windows.Forms.Keys.E Then
				searchControl1.Focus()
				e.IsInputKey = False
			End If
		End Sub
	End Class

End Namespace